scripts/start-dev-with-auto-restart.sh

```diff
@@ -0,0 +1,58 @@
+#!/bin/bash
+
+# Auto-restart webpack-dev-server on memory crashes
+# This script monitors for heap out of memory errors and automatically restarts
+# Usage: ./start-dev-with-auto-restart.sh [webpack-dev-server arguments]
+
+WDS_ARGS=("$@")
+MAX_RESTARTS=10
+RESTART_COUNT=0
+RESTART_DELAY=1
+
+echo "🚀 Starting webpack-dev-server with auto-restart (max restarts: $MAX_RESTARTS)"
+
+run_wds () {
+    # Check if platform is Desktop to determine open behavior
+    if [[ "${WDS_ARGS[*]}" == *"--env platform=desktop"* ]]; then
+        # For Desktop, always use --no-open since app is handled by Electron
+        node --expose-gc --max-old-space-size=1100 ./node_modules/.bin/webpack-dev-server --no-open "${WDS_ARGS[@]}" --config config/webpack/webpack.dev.ts
+    else
+        # For Web, use the provided open flag
+        node --expose-gc ./node_modules/.bin/webpack-dev-server "$1" "${WDS_ARGS[@]}" --config config/webpack/webpack.dev.ts
+    fi
+}
+
+while [ $RESTART_COUNT -lt $MAX_RESTARTS ]; do
```

Comment:
Just a bash style preference:

```suggestion
while [[ $RESTART_COUNT -lt $MAX_RESTARTS ]]; do
```